components/base/wheeled/wheeled_base.go

```diff
@@ -96,26 +107,49 @@ type wheeledBase struct {
 	right     []motor.Motor
 	allMotors []motor.Motor
 
-	opMgr  operation.SingleOperationManager
-	logger golog.Logger
+	sensors *sensors
+	// movementSensors   []movementsensor.MovementSensor
+	// orientationSensor movementsensor.MovementSensor
+
+	opMgr                   operation.SingleOperationManager
+	activeBackgroundWorkers *sync.WaitGroup
+	logger                  golog.Logger
+	cancelCtx               context.Context
+	cancelFunc              func()
 
 	name              string
 	collisionGeometry spatialmath.Geometry
 }
 
+type sensors struct {
+	ctx         context.Context
```

Comment:
if you mean the parent context from the PAI calls - I have tried to make them not cancel the for loop for the sensor - something is immediately canceling context as soon as runAll is called. So I spawned a next context to both keep the loop running, close the ticker when that context is done, and respawn once a new spin call with a movement sensor is started. 

If I'm mistaken in what you mean - please let me know.

Reply:
yeah, that wasn't what I meant. Spawning a new context is fine, I just mean you can store the channel for that context on the struct rather than the context itself